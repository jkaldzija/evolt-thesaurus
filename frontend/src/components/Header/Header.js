import React from 'react';
import style from "./header.module.css";
import {Container} from "reactstrap";
import classnames from "classnames";
import {Link, NavLink} from "react-router-dom";
import {routes} from "../Utility/routing";

class Header extends React.Component {

    render() {

        return <div className={classnames(style.wrapper, "d-flex align-items-center primary")}>
            <Container className={"d-flex align-items-center"}>
                <h1 className={classnames(style.title, "va-text")}>Thesaurus</h1>
                <NavLink exact activeClassName={style.active} className={style.link} to={routes.home.path}>Home</NavLink>
                <NavLink exact activeClassName={style.active} className={style.link} to={routes.dictionary.path}>Dictionary</NavLink>
                <Link className={classnames(style.link, "d-none d-md-block")} target={"_blank"} to={`/api/v1/swagger-ui.html`}>Docs</Link>
            </Container>
        </div>;
    };

}

export default Header;
