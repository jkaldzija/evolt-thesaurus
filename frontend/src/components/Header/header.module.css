.wrapper{
    height: 70px;
    margin-bottom: 40px;
}

.title{
    color: white;
    font-family: "Museo Sans Rounded 700";
    font-size: 25px;
}

.link{
    color: white !important;
    margin-left: 20px;

    font-family: "Museo Sans Rounded 500";
    padding: 5px 10px;
    border-radius: 20px;
}

.link:hover, .active{
    text-decoration: none !important;
    background-color: rgba(54, 54, 54, 0.08);
}

