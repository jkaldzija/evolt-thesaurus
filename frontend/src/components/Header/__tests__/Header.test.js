import React from 'react';
import {shallow} from "enzyme";
import Header from "../Header";

describe('Search Box', function () {
    it('should render correctly', function () {
        const component = shallow(
            <Header />
        );
        expect(component).toMatchSnapshot();
    });

});
