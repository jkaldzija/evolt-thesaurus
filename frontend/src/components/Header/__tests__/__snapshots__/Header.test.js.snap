// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render correctly 1`] = `
<div
  className="wrapper d-flex align-items-center primary"
>
  <Container
    className="d-flex align-items-center"
    tag="div"
  >
    <h1
      className="title va-text"
    >
      Thesaurus
    </h1>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="link"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="link"
      exact={true}
      to="/dictionary"
    >
      Dictionary
    </NavLink>
    <Link
      className="link d-none d-md-block"
      replace={false}
      target="_blank"
      to="/api/v1/swagger-ui.html"
    >
      Docs
    </Link>
  </Container>
</div>
`;
